packages/types/src/documents/app/table.ts

```diff
@@ -9,6 +9,10 @@ export enum RelationshipTypes {
   MANY_TO_MANY = "many-to-many",
 }
 
+export enum AutoReason {
+  FOREIGN_KEY = "foreign_key",
```

Comment:
This is a little bit confusing. Should we not have `primary_key` and other options?